scheduler/src/cook/mesos/api.clj

```diff
@@ -174,6 +175,13 @@
    (s/optional-key :extract) s/Bool
    (s/optional-key :cache) s/Bool})
 
+(def ComputeCluster
+  "Schema for a compute cluster"
+  {:compute-cluster-type s/Keyword
+   :compute-cluster-name s/Str
+   (s/optional-key :mesos-framework-id) s/Str
```

Comment:
I prefer for these kind of APIs to break them down by type, so if `compute-cluster-type` is mesos, have a separate `MesosComputeCluster` in the `ComputeCluster` under the `:mesos` key.

Reply:
Done.